membership_prorate/models/account_invoice.py

```diff
@@ -39,22 +40,34 @@ def _prepare_invoice_line_prorate_vals(self, invoice_line):
                 "date_from": date_invoice,
             }
 
-    @api.model
-    def create(self, vals):
-        invoice_line = super(AccountInvoiceLine, self).create(vals)
-        product = self.env["product.product"].browse(vals.get("product_id", False))
-        if not product.membership or not product.membership_prorate:
-            return invoice_line
-        # Change quantity accordingly the prorate
-        invoice_line_vals = self._prepare_invoice_line_prorate_vals(invoice_line)
-        if invoice_line_vals:
-            date_from = invoice_line_vals.pop("date_from")
-            invoice_line.write(invoice_line_vals)
-            # Rectify membership price and start date in this case
-            memb_line = self.env["membership.membership_line"].search(
-                [("account_invoice_line", "=", invoice_line.id)], limit=1
-            )
-            memb_line.write(
-                {"member_price": invoice_line.price_subtotal, "date_from": date_from}
-            )
-        return invoice_line
+    @api.model_create_multi
+    def create(self, vals_list):
+        invoice_lines = super().create(vals_list)
+        for invoice_line in invoice_lines.filtered(
+            lambda r: r.product_id.membership and r.product_id.membership_prorate
+        ):
+            # Change quantity accordingly the prorate
+            invoice_line_vals = self._prepare_invoice_line_prorate_vals(invoice_line)
+            if invoice_line_vals:
+                date_from = invoice_line_vals["date_from"]
+                quantity = invoice_line_vals["quantity"]
+                invoice = invoice_line.move_id
+                with Form(invoice) as invoice_form:
+                    for index in range(len(invoice.invoice_line_ids)):
+                        if (
+                            invoice_form.invoice_line_ids.edit(index)._values["id"]
+                            == invoice_line.id
+                        ):
+                            with invoice_form.invoice_line_ids.edit(index) as line_form:
+                                line_form.quantity = quantity
```

Comment:
```suggestion
                with Form(invoice) as invoice_form:
                    index = next((
                        index for (index, d) in enumerate(
                            invoice_form.invoice_line_ids._records
                        ) if d["id"] == invoice_line.id), None
                    )
                    if index is not None:
                        with invoice_form.invoice_line_ids.edit(index) as line_form:
                            line_form.quantity = quantity
```

Reply:
Thanks @chienandalu  , nice code by the way :D